### PR TITLE
Ability to enter cost as a string #6

### PR DESCRIPTION
**Pull request by petrenkonikita112263**
* change type from number into string for EstimatedCost field as cost property
* rename nbestimate into cost in the affacted files

### CODE REVIEW

File: application-ideas-ui/src/main/resources/Ideas/Migrations/1.14/WebHome.xml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+
+<xwikidoc version="1.3" reference="Ideas.Migrations.1\.14.WebHome" locale="">
```
Comment (oanalavinia) [inline application-ideas-ui/src/main/resources/Ideas/Migrations/1.14/WebHome.xml:23]: You should use terminal pages for code, so just `Ideas.Migrations.1\.14` here

File: application-ideas-ui/src/main/resources/Ideas/Migrations/MigrationMacros/WebHome.xml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+
+<xwikidoc version="1.3" reference="Ideas.Migrations.MigrationMacros.WebHome" locale="">
```
Comment (oanalavinia) [inline application-ideas-ui/src/main/resources/Ideas/Migrations/MigrationMacros/WebHome.xml:23]: Here as well, just `Ideas.Migrations.MigrationMacros`

Comment (oanalavinia) [reply application-ideas-ui/src/main/resources/Ideas/Migrations/MigrationMacros/WebHome.xml:23]: And were are you using this page? Note that the migration error should be displayed on the ideas homepage

File: application-ideas-ui/src/main/resources/Ideas/Migrations/1.14/WebHome.xml [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+      #set ($mDoc = $xwiki.getDocument($mDocRef))
+      #set ($ideasObj = $mDoc.getObject($ideasClass))
+      #set ($nbestimate = $ideasObj.getProperty('nbestimate').value)
+      #set ($saved = $NULL)
+      #if($nbestimate)
+        #set($stringCost = "$nbestimate")
+        ## Save the idea's cost value after converting to the string.
+        #set ($saved = 'Would be saved on actual run')
+        #if ("$!request.confirm" == 'true')
+          #set($discard = $ideasObj.set('cost', $stringCost))
```
Comment (oanalavinia) [inline application-ideas-ui/src/main/resources/Ideas/Migrations/1.14/WebHome.xml:69]: You need a space after `#set`

File: application-ideas-ui/src/main/resources/Ideas/Migrations/1.14/WebHome.xml [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+          #set ($discard = $mDoc.save('Move the idea''s estimatecost converted value to the cost'))
+          #set ($saved = 'Saved')
+        #end
+      #end
+      ## Clean the deprecated properties, if there are any
```
Comment (oanalavinia) [inline application-ideas-ui/src/main/resources/Ideas/Migrations/1.14/WebHome.xml:74]: Comments should always end with a `.`

File: application-ideas-ui/src/main/resources/Ideas/IdeasClass.xml [hunk 5]
```diff
@@ -584,8 +609,8 @@
           var nbv = response.opponents.length;
           var nbop = response.supporters.length;
         }
-        $('#pro-participants').load(XWiki.currentDocument.getURL('get') + ' #pro-participants>*', '');
-        $('#against-participants').load(XWiki.currentDocument.getURL('get') + ' #against-participants>*', '');
+        $('#pro-participants').load(XWiki.currentDocument.getURL('get') + ' #pro-participants&gt;*', '');
```
Comment (oanalavinia) [inline application-ideas-ui/src/main/resources/Ideas/IdeasClass.xml:612]: Why is this change needed?

File: application-ideas-ui/src/main/resources/Ideas/Migrations/WebHome.xml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+
+<xwikidoc version="1.3" reference="Ideas.Migrations.WebHome" locale="">
```
Comment (oanalavinia) [inline application-ideas-ui/src/main/resources/Ideas/Migrations/WebHome.xml:23]: You don't actually need to add this page too, since a parent is not necessarily needed in order to be able to have `Migrations.1.14` and `Migrations.MigrationMacros`

File: application-ideas-ui/src/main/resources/Ideas/Migrations/1.14/WebHome.xml [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+        #set ($discard = $xwiki.xWiki.saveDocument($documentToClean,
+          $services.localization.render('core.model.xobject.synchronizeObjects.versionSummary'), true, $xcontext.context))
+      #end
+      |[[$mDoc.prefixedFullName]]|$!saved|$cleanedProps
+    #end
+
+  #end
+#end
+
+#if ("$!request.confirm" != 'true')
```
Comment (oanalavinia) [inline application-ideas-ui/src/main/resources/Ideas/Migrations/1.14/WebHome.xml:99]: Can you add also a screenshot for how this page looks before and after a migration?

File: application-ideas-ui/src/main/resources/Ideas/Migrations/1.14/WebHome.xml [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+  <creator>xwiki:XWiki.Admin</creator>
+  <parent>Ideas.Migrations.WebHome</parent>
+  <author>xwiki:XWiki.Admin</author>
+  <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
+  <version>1.1</version>
+  <title>1.14</title>
+  <comment/>
+  <minorEdit>false</minorEdit>
+  <syntaxId>xwiki/2.1</syntaxId>
+  <hidden>false</hidden>
```
Comment (oanalavinia) [inline application-ideas-ui/src/main/resources/Ideas/Migrations/1.14/WebHome.xml:38]: This should be hidden

File: application-ideas-ui/src/main/resources/Ideas/Migrations/MigrationMacros/WebHome.xml [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+  <creator>xwiki:XWiki.Admin</creator>
+  <parent>Ideas.Migrations.WebHome</parent>
+  <author>xwiki:XWiki.Admin</author>
+  <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
+  <version>1.1</version>
+  <title>MigrationMacros</title>
+  <comment/>
+  <minorEdit>false</minorEdit>
+  <syntaxId>xwiki/2.1</syntaxId>
+  <hidden>false</hidden>
```
Comment (oanalavinia) [inline application-ideas-ui/src/main/resources/Ideas/Migrations/MigrationMacros/WebHome.xml:38]: This should be hidden

**Review by oanalavinia — CHANGES_REQUESTED** [review]

File: application-ideas-ui/src/main/resources/Ideas/Migrations/1.14/WebHome.xml (hunk 7 above)
Comment (petrenkonikita112263) [reply application-ideas-ui/src/main/resources/Ideas/Migrations/1.14/WebHome.xml:99]: Before
image
After
image

**Comment by petrenkonikita112263** [comment]
Close due to possible irrelevancy of the feature.
